lib/Conversion/TritonGPUToLLVM/ElementwiseOpToLLVM.cpp

```diff
@@ -80,6 +80,21 @@ const std::string Bf16_to_Fp8E5M2 =
     "or.b32 $0, nosign, sign;                    \n" // restore sign
     "}";
 
+const std::string Fp8E4M3Nv_to_Bf16 = "{                                      \n"
+    ".reg .b32 a<2>, b<2>;                  \n" // if input = 0xf1f2f3f4
+    "prmt.b32 a0, 0, $2, 0x5140;            \n" // a0 = 0xf300f400
+    "prmt.b32 a1, 0, $2, 0x7362;            \n" // a1 = 0xf100f200
+    "lop3.b32 b0, a0, 0x7fff7fff, 0, 0xc0;  \n" // b0 = a0 & 0x7fff7fff
+    "lop3.b32 b1, a1, 0x7fff7fff, 0, 0xc0;  \n" // (strip sign)
+    "shr.b32  b0, b0, 4;                    \n" // b0 >>= 4 to fill mantissa
+    "shr.b32  b1, b1, 4;                    \n" // shift into bf16 position
+    "add.u32  b0, b0, 0x3a003a00;           \n" // b0.exp += 127-11
+                                                // exponent compensate = 116
+    "add.u32  b1, b1, 0x3a003a00;           \n" // b1 += 116<<7 | 116<<7<<16
+    "lop3.b32 $0, b0, 0x80008000, a0, 0xf8; \n" // out0 = b0|(0x80008000&a0)
+    "lop3.b32 $1, b1, 0x80008000, a1, 0xf8; \n" // (restore sign)
+    "}";
```

Comment:
Since this format is only supported on hopper+ it would be more efficient to directly use `cvt.rn.satfinite.e4m3x2.f32` then shifting?

Reply:
I see. Why does the current `Fp8E5M2_to_Bf16` implementation not use `cvt`?

Reply:
mainly because it is also support on Ampere, we could use cvt for Hopper and have two version, I guess it has just not been needed.

Reply:
Agreed with @ThomasRaoux